packages/grafana-data/src/types/explore.ts

```diff
@@ -1,15 +1,28 @@
 import { DataQuery } from '@grafana/schema';
 
 import { PreferredVisualisationType } from './data';
-import { RawTimeRange, TimeRange } from './time';
+import { TimeRange } from './time';
 
 type AnyQuery = DataQuery & Record<string, any>;
 
+// enforce type-incompatibility with RawTimeRange to ensure it's parsed and converted.
+// URLRangeValue may be a string representing UTC time in ms, which is not a compatible
+// value for RawTimeRange when used as a string (it could only be an ISO formatted date)
+export type URLRangeValue = string | { __brand: 'URL Range Value' };
```

Comment:
what is the purpose of `{__brand: 'URL Range Value' }`?

Reply:
It makes it impossible to cast URLRange directly to RawTimeRange:

before (no type error when converting from URLRange to RawTimeRange):

<img width="1213" alt="Screenshot 2023-08-16 at 12 32 13" src="https://github.com/grafana/grafana/assets/745532/18476ac7-0b52-4240-a76e-1882c1e702c4">

after:

<img width="1388" alt="Screenshot 2023-08-16 at 12 32 28" src="https://github.com/grafana/grafana/assets/745532/da65cb35-af3f-45bf-ab0d-35515188e4ed">